### PR TITLE
initial commit

### PR DESCRIPTION
**Pull request by sayali-salunkhe**

### CODE REVIEW

File: src/main/java/uk/gov/hmcts/reform/refunds/services/IdamServiceImpl.java [hunk 1]
```diff
@@ -33,6 +36,9 @@
     public static final String USER_FULL_NAME_ENDPOINT = "/api/v1/users";
     private static final Logger LOG = LoggerFactory.getLogger(IdamServiceImpl.class);
     static private final String LIBERATA_NAME = "Middle office provider";
+    private static final String INTERNAL_SERVER_ERROR = "Internal Server error. Please, try again later";
```
Comment (sathishkannan-hmcts) [inline src/main/java/uk/gov/hmcts/reform/refunds/services/IdamServiceImpl.java:39]: Change the error message to be specific as "User details not found for these roles in IDAM"

File: src/main/java/uk/gov/hmcts/reform/refunds/services/IdamServiceImpl.java [hunk 2]
```diff
@@ -33,6 +36,9 @@
...
+    private static final int USER_INFO_SIZE = 100;
```
Comment (sathishkannan-hmcts) [inline src/main/java/uk/gov/hmcts/reform/refunds/services/IdamServiceImpl.java:40]: Move this to yaml file and change the value to 300

File: src/main/java/uk/gov/hmcts/reform/refunds/services/IdamServiceImpl.java [hunk 3]
```diff
@@ -33,6 +36,9 @@
...
+    static private final String LAST_MODIFIED_TIME = ">now-720d";
```
Comment (sathishkannan-hmcts) [inline src/main/java/uk/gov/hmcts/reform/refunds/services/IdamServiceImpl.java:41]: Keep only the last unit (720d) and move to yaml file

File: src/main/java/uk/gov/hmcts/reform/refunds/services/IdamServiceImpl.java [hunk 4]
```diff
@@ -105,27 +111,62 @@ public UserIdentityDataDto getUserIdentityData(MultiValueMap<String, String> hea
...
+    @Override
+    public Set<String> getUserIdSetForService(MultiValueMap<String, String> headers, List<String> roles) {
+        /*Set<String> users = new HashSet<>();
+        users.add("asdfghjk-kjhgfds-dfghj-sdfghjk");
+        return users;*/
+
+//        Pattern rolePattern = Pattern.compile("^.*damage.*$");
+
+        UriComponentsBuilder builder = UriComponentsBuilder.fromUriString(idamBaseURL + USER_FULL_NAME_ENDPOINT)
+                .queryParam("query", "(roles:" + roles + ") AND lastModified:" + LAST_MODIFIED_TIME)
```
Comment (sathishkannan-hmcts) [inline src/main/java/uk/gov/hmcts/reform/refunds/services/IdamServiceImpl.java:146]: add >now- to the lastmodified in the query

File: src/main/java/uk/gov/hmcts/reform/refunds/services/RefundsServiceImpl.java [hunk 5]
```diff
@@ -109,16 +110,41 @@ public RefundListDtoResponse getRefundList(String status, MultiValueMap<String,
...
 
+    private Optional<List<Refund>> filterRefundList(Optional<List<Refund>> refundList, MultiValueMap<String, String> headers,
+                                                    List<String> roles) {
+        Set<String> distintUserIDSet = idamService.getUserIdSetForService(headers, roles);
+
+        /*refundList.ifPresent(list -> list.stream()
+                .filter(refunds -> distintUserIDSet.contains(refunds.getCreatedBy()))
+                .collect(Collectors.toList()));*/
+
+        List<Refund> finalList = new ArrayList<>();
```
Comment (sathishkannan-hmcts) [inline src/main/java/uk/gov/hmcts/reform/refunds/services/RefundsServiceImpl.java:135]: Suggestion: Change this to stream if possible

**Review by sathishkannan-hmcts — CHANGES_REQUESTED** [review]

File: src/main/java/uk/gov/hmcts/reform/refunds/services/IdamServiceImpl.java (hunk 1 above)
Comment (sayali-salunkhe) [reply src/main/java/uk/gov/hmcts/reform/refunds/services/IdamServiceImpl.java:39]: Done

File: src/main/java/uk/gov/hmcts/reform/refunds/services/IdamServiceImpl.java (hunk 2 above)
Comment (sayali-salunkhe) [reply src/main/java/uk/gov/hmcts/reform/refunds/services/IdamServiceImpl.java:40]: Done

File: src/main/java/uk/gov/hmcts/reform/refunds/services/IdamServiceImpl.java (hunk 3 above)
Comment (sayali-salunkhe) [reply src/main/java/uk/gov/hmcts/reform/refunds/services/IdamServiceImpl.java:41]: Done

File: src/main/java/uk/gov/hmcts/reform/refunds/services/IdamServiceImpl.java (hunk 4 above)
Comment (sayali-salunkhe) [reply src/main/java/uk/gov/hmcts/reform/refunds/services/IdamServiceImpl.java:146]: Done

File: src/main/java/uk/gov/hmcts/reform/refunds/services/RefundsServiceImpl.java (hunk 5 above)
Comment (sayali-salunkhe) [reply src/main/java/uk/gov/hmcts/reform/refunds/services/RefundsServiceImpl.java:135]: Done

**Review by sathishkannan-hmcts — APPROVED** [review]

**Pull request merged**